src/Tests/dotnet-new.Tests/CommonTemplatesTests.cs

```diff
@@ -323,468 +150,121 @@ public void NuGetConfigPermissions()
-                .ExitWith(0)
-                .And.NotHaveStdErr();
-
-            CommandResult buildResult = new DotnetBuildCommand(_log, "MyProject")
-                .WithWorkingDirectory(workingDir)
-                .Execute();
-
-            if (buildPass)
-            {
-                buildResult.Should().ExitWith(0).And.NotHaveStdErr();
-            }
-            else
-            {
-                buildResult.Should().Fail();
-                return;
-            }
-            string codeFileName = name == "console" ? "Program.cs" : "Class1.cs";
-            string programFileContent = File.ReadAllText(Path.Combine(workingDir, "MyProject", codeFileName));
-            XDocument projectXml = XDocument.Load(Path.Combine(workingDir, "MyProject", "MyProject.csproj"));
-            XNamespace ns = projectXml.Root?.Name.Namespace ?? throw new Exception("Unexpected project file format");
-            if (supportsFeature)
-            {
-                Assert.DoesNotContain("using System;", programFileContent);
-                Assert.Equal("enable", projectXml.Root?.Element(ns + "PropertyGroup")?.Element(ns + "ImplicitUsings")?.Value);
-            }
-            else
-            {
-                Assert.Contains("using System;", programFileContent);
-                Assert.Null(projectXml.Root?.Element(ns + "PropertyGroup")?.Element(ns + "ImplicitUsings"));
-            }
-        }
-
-        public static IEnumerable<object?[]> FileScopedNamespacesSupport_Data()
+        public async void FeaturesSupport(
+            string name,
+            bool buildPass,
+            string? framework,
+            string? langVersion,
+            bool langVersionUnsupported,
+            string? language,
+            bool supportsNullable,
+            bool supportsTopLevel,
+            bool forceDisableTopLevel,
+            bool supportsImplicitUsings,
+            bool supportsFileScopedNs)
         {
-            var templatesToTest = new[]
-            {
-                new { Template = "classlib", Frameworks = new[] { null, "net7.0", "netstandard2.0", "netstandard2.1" } }
-            };
-            string[] unsupportedLanguageVersions = { "1", "ISO-1" };
-            string?[] supportedLanguageVersions = { null, "ISO-2", "2", "3", "4", "5", "6", "7", "7.1", "7.2", "7.3", "8.0", "9.0", "10.0", "11.0", "latest", "latestMajor", "default", "preview" };
+            const string currentDefaultFramework = "net7.0";
+            //string currentDefaultFramework = $"net{Environment.Version.Major}.{Environment.Version.Minor}";
 
-            string?[] supportedFrameworks = { null, "net7.0" };
-            string?[] fileScopedNamespacesSupportedLanguages = { "10.0", "11.0", "latest", "latestMajor", "default", "preview" };
+            string workingDir = CreateTemporaryFolder(folderName: $"{name}-{langVersion ?? "null"}-{framework ?? "null"}");
+            string outputDir = "MyProject";
+            string projName = name;
```

Comment:
Yes - it's another way. 
I'd prefer to avoid it though as it would make snapshots naming less descriptive (e.g. "TestProject.csproj" instead of "classlib.csproj")